src/jit/function-builder.cc

```diff
@@ -76,6 +76,64 @@ inline Opcode ReadOpcodeAt(const uint8_t* pc) {
   return ReadOpcode(&pc);
 }
 
+#define CHECK_TRAP_IN_HELPER(...)                \
+  do {                                           \
+    wabt::interp::Result result = (__VA_ARGS__); \
+    if (result != wabt::interp::Result::Ok) {    \
+      return static_cast<Result_t>(result);      \
+    }                                            \
+  } while (0)
+
+FunctionBuilder::Result_t FunctionBuilder::CallHelper(wabt::interp::Thread* th, wabt::interp::IstreamOffset offset) {
+  // no need to check if JIT was enabled since we can only get here it was
+  auto meta_it = th->env_->jit_meta_.find(offset);
+
+  auto call_interp = [&](){
+    th->set_pc(offset);
+    auto last_jit_frame = th->last_jit_frame_;
+    th->last_jit_frame_ = th->pc_;
+    th->set_pc(offset);
+    wabt::interp::Result result = wabt::interp::Result::Ok;
+    while (result == wabt::interp::Result::Ok) {
+      result = th->Run(1000);
+    }
+    th->last_jit_frame_ = last_jit_frame;
+    return result;
+  };
+
+  auto currnt_pc = &(th->GetIstream()[th->pc_]);
+  CHECK_TRAP_IN_HELPER(th->PushCall(currnt_pc));
+  if (meta_it != th->env_->jit_meta_.end()) {
+    auto meta = &meta_it->second;
+    if (!meta->tried_jit) {
+      meta->num_calls++;
+
+      if (meta->num_calls >= th->env_->jit_threshold) {
+        meta->jit_fn = jit::compile(th, meta->wasm_fn);
+        meta->tried_jit = true;
+
+        if (th->env_->trap_on_failed_comp && meta->jit_fn == nullptr)
+          return static_cast<Result_t>(wabt::interp::Result::TrapFailedJITCompilation);
+      }
+    }
+
+    if (meta->jit_fn) {
+      CHECK_TRAP_IN_HELPER(meta->jit_fn());
+    } else {
+      auto result = call_interp();
+      if (result != wabt::interp::Result::Returned)
+        return static_cast<Result_t>(result);
+    }
+  } else {
+    auto result = call_interp();
+    if (result != wabt::interp::Result::Returned)
+      return static_cast<Result_t>(result);
+  }
+  th->set_pc(th->PopCall());
```

Comment:
The `th->set_pc` is unnecessary here, since the value in there will never be used.

Reply:
Initially, I wanted that call to be there to "perfectly-mimic" the behaviour of the interpreter. However, since we're already not doing that in other places, I'll get rid of it.